src/Benchmarks/Program.cs

```diff
@@ -186,5 +187,21 @@ private static void Listen(KestrelServerOptions options, IConfigurationRoot conf
                 }
             });
         }
+
+        private static IPEndPoint CreateIPEndPoint(Uri uri)
+        {
+            IPAddress ip;
+
+            if (string.Equals(uri.Host, "localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                ip = IPAddress.Loopback;
```

Comment:
What if the port is also 0? Do we add logic to throw in that case like we do Kestrel? It also complicates things slightly (but not too much) to have multiple IPEndpoints per URL.

In practice IPv4 loopback always seems to work when "localhost" is specified, but I'll change it if it's important